library/options.inc.php

```diff
@@ -3159,5 +3160,32 @@ function lbf_current_value($frow, $formid, $encounter) {
   }
   return $currvalue;
 }
-
+# Generates a dropdown list for providers. This Code added by Terry Hill teryhill@librehealth.io
+# If you specify the $allprov it will list everyone that has an entry in the NPI field.
+function genProviderSelect($selname, $toptext, $default=0, $disabled=false, $allprov=false) 
+{
+  if($allprov == 1) {
+  $query = "SELECT id, lname, mname, fname FROM users WHERE " .
+    "( authorized = 1 OR npi != '' ) " .
+    "AND active = 1 AND ( info IS NULL OR info NOT LIKE '%Inactive%' ) " .
+    "ORDER BY lname, fname";
+  } else { 
+     $query = "SELECT id, lname, mname, fname FROM users WHERE " .
```

Comment:
I think that using "LIKE" in here is a very bad idea predicated on people using particular text.  There is a provider type available, authorized, and active.  Ensuring that the provider types selection in the user admin or in the address book or other GUI's allow the user to clearly say "this is a provider of some type of service", and using the other active and authorized status markers are what we should go with.  I have found text like "Her Medicaid License Was Inactive During Last March" in a user info box before, which messed lots of things up in a very mysterious fashion.